### PR TITLE
feat(fclasses): Overload asTokenSpec.

### PR DESCRIPTION
**Pull request by wodenx**
## Changes
- Allow asTokenSpec to take className strings or HOC's which are applied to _ key of Core domain
- Provide a single 'Core' domain as a default when asTokenSpec is invoked with no arguments.

### CODE REVIEW

File: packages/fclasses/__tests__/tokenSpec.test.tsx [hunk 1]
```diff
@@ -87,6 +87,48 @@ describe('extendMeta', () => {
...
+    });
+    expect(test.Core).toBeDefined();
+    // Check that core domain and default domains ar allowed
+    const test1 = asTokenSpec()({
+      Core: {
+        _: 'test',
+      },
+      Flow: flowHoc,
+      Compose: {},
+    });
```
Comment (dewen) [inline packages/fclasses/__tests__/tokenSpec.test.tsx:104]: Do we expect test1.Core to be 'test' here?

File: packages/fclasses/__tests__/tokenSpec.test.tsx [hunk 2]
```diff
@@ -87,6 +87,48 @@ describe('extendMeta', () => {
...
 
 describe('asTokenSpec', () => {
+  it('Supplies core domain by default', () => {
+    const test = asTokenSpec()({
+      Meta: {
+        title: 'Test',
+      },
+    });
+    expect(test.Core).toBeDefined();
+    // Check that core domain and default domains ar allowed
```
Comment (dewen) [inline packages/fclasses/__tests__/tokenSpec.test.tsx:97]: ar => are

File: packages/fclasses/src/tokenSpec.ts [hunk 3]
```diff
@@ -249,7 +288,11 @@ const asTokenSpec = <
     spec.Compose = {};
     spec.Flow = undefined;
     spec.Meta = {};
-    mergeWith(spec, ...specs, tokenMergeCustomizer);
+    const coreDomain = Object.keys(d)[0];
```
Comment (dewen) [inline packages/fclasses/src/tokenSpec.ts:291]: Are we sure the first key is always core domain? I thought user could provide different keyed domains

Comment (wodenx) [reply packages/fclasses/src/tokenSpec.ts:291]: It will use whatever the first key is.  There is no guarantee that there is a 'Core' domain at all.

**Review by dewen — DISMISSED** [review]

File: packages/fclasses/src/tokenSpec.ts [hunk 4]
```diff
@@ -222,24 +222,63 @@ const on = (
...
+ * If specified, defines a circumscribed set of allowed token domains.  If omitted, a single
+ * 'Core' domain will be allowed.  The resulting function will produce a token specification
+ * in which all domains are guaranteed to be present (those which were not defined will be
+ * empty objects). The order of the domains is consistent and defined by the order of the keys
+ * in this parameter.
  *
  * @returns
- * Function which recieves an object as a parameter and returns a normalized token
- * specification based on that object.
+ * Function which receives a set of partial token speciications and returnsa normalized token
```
Comment (marcopagliarulo) [inline packages/fclasses/src/tokenSpec.ts:243]: Please add a space between returns and a. Fix the typo in specifications

**Review by marcopagliarulo — DISMISSED** [review]

File: packages/fclasses/src/tokenSpec.ts (hunk 4 above)
Comment (wodenx) [inline packages/fclasses/src/tokenSpec.ts:243]:
```suggestion
 * Function which receives a set of partial token speciications and returns a normalized token
```

**Review by marcopagliarulo — APPROVED** [review]

**Review by dewen — APPROVED** [review]

**Pull request merged**